### PR TITLE
Feat/detect line text overlap

### PR DESCRIPTION
**Pull request by ethan-zhx**
## Summary
Improve the Slides XML layout lint gate by tightening canvas overflow reporting and adding detection for visible line elements that cross readable text glyphs.

## Changes
- Report ghost/faint text shapes when they overflow the slide canvas, while still allowing their intentional decorative overlaps.
- Detect `<line>` elements that cross text glyphs and report them as `bbox_overlap` errors.
- Use segment-level geometry for horizontal, vertical, and diagonal lines to avoid naive bounding-box false positives.
- Ignore non-actionable cases such as invisible lines, frame-edge grazing, padding-only intersections, polylines, and lines outside the actual visual glyph area.
- Add regression coverage for ghost text canvas overflow and line/text crossing edge cases.

## Test Plan
- [ ] Unit tests pass
- [ ] Manual local verification confirms the `lark-cli slides` lint flow works as expected

## Related Issues
- None

## Summary by CodeRabbit

* **New Features**
  * Added lint checks that detect visible lines crossing text glyphs.
  * Improved detection accuracy by ignoring nearly invisible lines, frame-only contact, and near-edge grazes.
* **Bug Fixes**
  * Out-of-canvas text is now reported consistently, including low-opacity text.
  * Reduced false positives for diagonal lines and lines outside visible glyph areas.
* **Tests**
  * Expanded coverage for horizontal, vertical, diagonal, and exempt line/text scenarios.

### CODE REVIEW

File: skills/lark-slides/scripts/xml_text_overlap_lint.py [hunk 1]
```diff
@@ -1500,8 +1507,6 @@ def detect_elements_out_of_canvas(
         if element["kind"] in {"table", "chart"}
         or (element["kind"] == "shape" and element["type"] in {"rect", "text"})
     ):
-        if is_ghost_text(element):
```
Comment (R0bynZhu) [inline skills/lark-slides/scripts/xml_text_overlap_lint.py:1503]: 为什么去掉了

Comment (ethan-zhx) [reply skills/lark-slides/scripts/xml_text_overlap_lint.py:1503]: 经过确认，幽灵字还是检测画布溢出比较合适

File: skills/lark-slides/scripts/xml_text_overlap_lint.py [hunk 2]
```diff
@@ -1611,6 +1616,93 @@ def detect_table_layout_size_mismatches(elements: list[dict[str, Any]]) -> list[
...
+def erode_rect(rect: dict[str, int | float], margin: float) -> dict[str, int | float] | None:
+    width = rect["width"] - 2 * margin
+    height = rect["height"] - 2 * margin
+    if width <= 0 or height <= 0:
+        return None
+    return {"x": rect["x"] + margin, "y": rect["y"] + margin, "width": width, "height": height}
+
+
+def line_crosses_text(line: dict[str, Any], text_element: dict[str, Any]) -> bool:
+    if not is_visually_rendered(line) or line.get("alpha", 1) < LINE_MIN_VISIBLE_ALPHA:
```
Comment (BD-ZERO) [inline skills/lark-slides/scripts/xml_text_overlap_lint.py:1663]: 原有的is_visually_rendered函数带有line的alpha判断，是否应该要合入到一起，可见性逻辑更集中一点

**Review by R0bynZhu — APPROVED** [review]

File: skills/lark-slides/scripts/xml_text_overlap_lint.py [hunk 3]
```diff
@@ -2244,8 +2346,12 @@ def extract_line_elements(slide_xml: str) -> list[dict[str, Any]]:
                 "y": min(start_y, end_y),
                 "width": abs(end_x - start_x),
                 "height": abs(end_y - start_y),
+                "startX": start_x,
+                "startY": start_y,
+                "endX": end_x,
+                "endY": end_y,
                 "rotation": 0,
-                "alpha": line_alpha if line_alpha is not None else 1,
+                "alpha": base_alpha * border_alpha,
```
Comment (fangshuyu-768) [inline skills/lark-slides/scripts/xml_text_overlap_lint.py:2354]: The effective border width is not preserved here, so the crossing test later evaluates only the center segment. This makes a line with border width="0" produce a blocking overlap, while a thick stroke whose centerline stays just outside the eroded rectangle is missed even though its rendered pixels cover the text. Please extract the effective width (defaulting to 2), ignore non-rendered zero-width strokes, and test the stroked segment/capsule rather than only its centerline.

File: skills/lark-slides/scripts/xml_text_overlap_lint.py [hunk 4]
```diff
@@ -1611,6 +1616,93 @@ def detect_table_layout_size_mismatches(elements: list[dict[str, Any]]) -> list[
...
+
+
+def line_crosses_text(line: dict[str, Any], text_element: dict[str, Any]) -> bool:
+    if not is_visually_rendered(line) or line.get("alpha", 1) < LINE_MIN_VISIBLE_ALPHA:
+        return False
+    if not is_text_element(text_element) or not has_text_content(text_element):
+        return False
+    if is_ghost_text(text_element) or is_decorative_text(text_element):
+        return False
+    glyph_bbox = estimate_text_visual_bbox(text_element)
```
Comment (fangshuyu-768) [inline skills/lark-slides/scripts/xml_text_overlap_lint.py:1669]: estimate_text_visual_bbox returns one unrotated rectangle for the entire text block. Consequently, crossings through rotated or vertical text are missed, while a line passing through the empty right side of a short second line is reported because a longer first line widens the shared rectangle. Since this is a blocking glyph-crossing rule, please evaluate per-line visual rectangles and apply rotation/vertical-text transforms, or skip/downgrade cases whose geometry is not modeled reliably.

File: skills/lark-slides/scripts/xml_text_overlap_lint.py [hunk 5]
```diff
@@ -1611,6 +1616,93 @@ def detect_table_layout_size_mismatches(elements: list[dict[str, Any]]) -> list[
...
+        return False
+    return segment_intersects_rect(
+        line["startX"], line["startY"], line["endX"], line["endY"], target
+    )
+
+
+def detect_line_text_crossings(
+    slide_xml: str, elements: list[dict[str, Any]]
+) -> list[dict[str, Any]]:
+    lines = extract_line_elements(slide_xml)
```
Comment (fangshuyu-768) [inline skills/lark-slides/scripts/xml_text_overlap_lint.py:1685]: Re-parsing lines separately from the text elements loses their shared XML draw order and occlusion relationship. A line placed before a later text shape with an opaque fill is completely hidden in the rendered slide, but this detector still emits a blocking overlap. Please preserve unified DOM order and exclude line portions hidden by later opaque shapes before classifying the line as visibly crossing text.

File: skills/lark-slides/scripts/xml_text_overlap_lint.py [hunk 6]
```diff
@@ -1611,6 +1616,93 @@ def detect_table_layout_size_mismatches(elements: list[dict[str, Any]]) -> list[
...
+    text_elements = [element for element in elements if is_text_element(element)]
+    issues: list[dict[str, Any]] = []
+    for line in lines:
+        for text_element in text_elements:
+            if not line_crosses_text(line, text_element):
+                continue
+            issues.append(
+                {
+                    "level": "error",
+                    "code": "bbox_overlap",
```
Comment (fangshuyu-768) [inline skills/lark-slides/scripts/xml_text_overlap_lint.py:1697]: Reusing bbox_overlap makes the normalized diagnostic contradict itself. For a horizontal line, the stored line bbox has height 0, so issue_measurement reports intersection_area=0 even though this branch emits an error, while the associated rule says intersection_area == 0 is the passing condition. Please introduce a dedicated line_text_crossing code with segment/stroke measurements, or attach an explicit measurement representing the geometry used by line_crosses_text.

**Review by fangshuyu-768 — COMMENTED** [review]
The 124 existing tests pass, but targeted cases exposed four correctness issues in the new blocking line/text crossing rule.

File: skills/lark-slides/scripts/xml_text_overlap_lint.py (hunk 2 above)
Comment (ethan-zhx) [reply skills/lark-slides/scripts/xml_text_overlap_lint.py:1663]: line的alpha值后续还要调，不合并is_visually_rendered。

File: skills/lark-slides/scripts/xml_text_overlap_lint.py (hunk 4 above)
Comment (ethan-zhx) [reply skills/lark-slides/scripts/xml_text_overlap_lint.py:1669]: 漏报问题，下一次迭代处理。

Comment (ethan-zhx) [reply skills/lark-slides/scripts/xml_text_overlap_lint.py:1669]: 旋转/竖排暂时无法可靠建模
